### DISCUSSION TITLE
SEO - Sitemap - Had to republish ALL content definitions to get the custom URLs in the sitemap

### DISCUSSION THREAD

**Discussion by vengi83644** [OrchardCMS/OrchardCore#10793] [Q&A]
We are having many content types and definitions for them. We are trying to enable the sitemap by enabling the AutoRoute feature.

Here is how I have added the custom URL in the AutoRoute part of a parent content type.
![image](https://user-images.githubusercontent.com/5276778/144380633-e6109943-3837-4c1c-93d4-50be72418c6f.png)

But when I generate the sitemap, it still shows me the default URL something like - /Contents/ContentItems/4vzrrcbtphfth668qs0z1gddnj

When I go to the content definition, it shows the permalink as empty.
![image](https://user-images.githubusercontent.com/5276778/144380848-2a46372e-dea6-4234-b19d-af6ca01b4835.png)

If I publish this now, I can the custom URL and also in the sitemap. We don't want to do muscle work for re-publishing all our 1000s of articles. 

Is there a way to get this sorted out? or Is there anything I am missing?

Thanks,
R. Venkatesan

**Comment by vengi83644** [comment]
Is there an automation to publish the contents?

**Reply by Skrypt** [reply]
There is no automation as of now but I think this would be a good idea to be able to repopulate these Alias. The issue here is that re-publishing all these content items all at once is probably not a light task. So, one way to do this is with SQL and then restart the application. Needs some thoughts but this discussion should be definitely moved to an issue - enhancement request.
